tensorflow/compiler/jit/mark_for_compilation_pass.cc

```diff
@@ -892,6 +892,23 @@ Status MarkForCompilationPassImpl::FindCompilationCandidates() {
       continue;
     }
 
+    //
+    // This is to workaround a problem with the following use pattern,
+    //     op0 : VarHandleOp
```

Comment:
I'm not sure when this can actually happen -- I'd expect deadness analysis to block putting `op2` and `op3` into the same cluster.

Reply:
It is hard to produce a small test case for this since it involves Switch and resources.  I will see 
if I could create a list of nodes in a single cluster which exposes the behavior.

Reply:
I dumped the clusters and related Switch node from the real case as follows.
It appears the related nodes are AssignVariableOp (3722) and ReadVariableOp (2719), I will update
the comments in the PR.

Switch Node: 1703 : cond_1/ReadVariableOp/Switch
  op: Switch
  input:  (0) 429 : global_step : VarHandleOp
  input:  (0) 78 : bert/embeddings/assert_less_equal/All : Const
  output:  (0) 2719 : cond_1/ReadVariableOp : ReadVariableOp

Cluster: cluster_1 contains the following nodes,

Node: 428 : cond_1/add/y
  op: Const
  input-control:  (-1) 0 : _SOURCE : NoOp
  output:  (1) 3583 : cond_1/add : Add
Node: 2719 : cond_1/ReadVariableOp
  op: ReadVariableOp
  input:  (1) 1703 : cond_1/ReadVariableOp/Switch : Switch
  output:  (0) 3583 : cond_1/add : Add
Node: 3583 : cond_1/add
  op: Add
  input:  (0) 2719 : cond_1/ReadVariableOp : ReadVariableOp
  input:  (0) 428 : cond_1/add/y : Const
  output:  (0) 3665 : step_update : Identity
Node: 3665 : step_update
  op: Identity
  input:  (0) 3583 : cond_1/add : Add
  output:  (1) 3722 : AssignVariableOp : AssignVariableOp
  output:  (0) 14123 : _retval_step_update_0_2 : _Retval
Node: 3722 : AssignVariableOp
  op: AssignVariableOp
  input:  (0) 429 : global_step : VarHandleOp
  input:  (0) 3665 : step_update : Identity
  output-control:  (-1) 14120 : group_deps_1 : NoOp
Node: 14120 : group_deps_1
  op: NoOp
  input-control:  (-1) 11657 : Assign_1536 : Assign

Reply:
I see, so the graph is http://bit.ly/2Lu5aoQ which is why Add and AssignVariable both have the same deadness.  Can you please add this graph as a comment (using the graphviz notation)?